src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs

```diff
@@ -181,14 +181,27 @@ await _dataImportService.UpdateStatus(dataImport.Id,
                             .Select(f => f.AutoSelectFilterItemLabel)
                             .SingleOrDefault() ?? "Total"; // If meta file didn't specify a default, look for "Total"
 
-                        return filterItems
+                        var filterItemsWithAutoSelectLabel = filterItems
                             .Where(item =>
-                                item.FilterGroup.FilterId == filterId
-                                && item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase))
-                            .Select(item => new { item.Id , item.Label})
-                            // There might be two filter items with the same label under different groups.
-                            // If so, we set no autoSelectFilterItem.
-                            .SingleOrDefault();
+                                item.FilterGroup.FilterId == filterId &&
+                                item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase));
+
+                        // There might be two or more filter items with the same label under different groups
+                        if (filterItemsWithAutoSelectLabel.Count() > 1)
+                        {
+                            // If so, does one belong to a filter group with the same label?
+                            var filterItemWithMatchingGroupLabel = filterItemsWithAutoSelectLabel.SingleOrDefault(fi => fi.FilterGroup.Label == autoSelectFilterItemLabel);
+
+                            if (filterItemWithMatchingGroupLabel is not null)
+                            {
+                                // If it does, this is the auto select filter item
+                                return new { filterItemWithMatchingGroupLabel.Id, filterItemWithMatchingGroupLabel.Label };
+                            }
+                        }
+
+                        return filterItemsWithAutoSelectLabel.Any()
+                            ? new { filterItemsWithAutoSelectLabel.First().Id, filterItemsWithAutoSelectLabel.First().Label }
+                            : null;
```

Comment:
Using `Any` like this actually has the following linter warning:

> CA1860: Prefer comparing 'Count' to 0 rather than using 'Any()', both for clarity and for performance

We're also doing a repeated `First` calls, which is a bit redundant too. Think we could tighten this up to the following:

```cs
return filterItemsWithAutoSelectLabel
    .Select(fi => new { fi.Id, fi.Label })
    .FirstOrDefault();
```